flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumerator.java

```diff
@@ -163,9 +180,51 @@ public KinesisStreamsSourceEnumeratorState snapshotState(long checkpointId) thro
 
     @Override
     public void close() throws IOException {
+        efoConsumerCleanup(sourceConfig, streamConsumerRegistrar);
         streamProxy.close();
     }
 
+    private void efoConsumerStartSetup(
```

Comment:
It appears that this code is fairly independent from the rest of this class. As per single responsibility principle, can we move this code away to a dedicated class and then think about the best way to plug it into the Enumerator lifecycle?

Reply:
Refactored logic into existing `StreamConsumerRegistrar`